1-foundations/1-types-and-syntax/foundations-s1.ipynb

```diff
@@ -1,11 +1,1580 @@
+    "print(list_int[3:8])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Important error message: IndexError**\n",
+    "    \n",
+    "Whenever you see an error where it says \"index out of range\", as in `IndexError: list index out of range`, then it means that you have tried to access an item in the list, using an index that is not used in the list."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# IndexError: list index out of range\n",
+    "print(list_int[10])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Edit a list:**\n",
+    "\n",
+    "So far, every time we have modified a variable we have used a `=`. For example `x = x + 1` or `name = list_mix[1]`. \n",
+    "\n",
+    "Lists have some _in-place_ operator methods, meaning methods that change modify the item itself.\n",
+    "\n",
+    "You find a list of more list methods here: https://www.w3schools.com/python/python_ref_list.asp"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a list of strs\n",
+    "pets = [\"cat\",\"dog\"]\n",
+    "print('Variable pets:', pets, type(pets))\n",
+    "\n",
+    "# Add one item to the list - .append() is an in-place operator\n",
+    "pets.append(\"gold fish\")\n",
+    "print('Variable pets:', pets, type(pets))\n",
+    "\n",
+    "# Note that we did not do: pets = pets.append(\"gold fish\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add another item to the list using in-place .append() and the \"=\" assign operator\n",
+    "pets_append_return = pets.append(\"butterfly\")\n",
+    "print('Variable pets:', pets, type(pets))\n",
+    "print('Variable pets_append_return:', pets_append_return, type(pets_append_return))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Re-create the list of pets\n",
+    "pets = [\"cat\", \"dog\", \"gold fish\", \"butterfly\"]\n",
+    "\n",
+    "# Print item with index 3 in original list\n",
+    "print('Print pet at index 3:', pets[3], type(pets[3]))\n",
+    "\n",
+    "# Add item at index 2\n",
+    "pets.insert(2,\"parrot\")\n",
+    "\n",
+    "# Print item with index 3 again\n",
+    "print('Print pet at index 3:', pets[3], type(pets[3]))\n",
+    "\n",
+    "#Print all pets\n",
+    "print('Variable pets:', pets, type(pets))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Modify item with index 1 \n",
+    "pets[1] = \"wolf\"\n",
+    "print('Variable pets:', pets, type(pets))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Work with lists:**"
```

Comment:
I suggest adding the times (`*`) operator with lists

Reply:
done in: 360840516a909edfe6038351da81f415d8e85b2f